File: Final_Checkpoint/Models/Vault.cs
namespace Final_Checkpoint.Models;

public class Vault : RepoItem<int>
{
  public string Name { get; set; }
  public string Description { get; set; }
  public string Img { get; set; }
  public bool IsPrivate { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
}

File: Final_Checkpoint.client/src/models/Vault.js
import { Profile } from "./Account.js"
import { RepoItem } from "./RepoItem.js"

export class Vault extends RepoItem {
  constructor(data) {
    super(data)
    this.name = data.name
    this.description = data.description
    this.img = data.img
    this.isPrivate = data.isPrivate
    this.creatorId = data.creatorId
    this.creator = new Profile(data.creator)
  }
}